packages/story-editor/src/components/colorPicker/basicColorList.js

```diff
@@ -100,54 +99,56 @@ function BasicColorList({
     : __('global', 'web-stories');
   return (
     <SwatchList ref={listRef} {...rest}>
-      {colors.map((pattern, i) => {
-        const isTransparentAndInvalid = !allowsOpacity && hasOpacity(pattern);
-        const isGradientAndInvalid = !allowsGradient && hasGradient(pattern);
-        const isDisabled =
-          (isTransparentAndInvalid || isGradientAndInvalid) && !isEditMode;
-        let tooltip = null;
-        if (isDisabled && !isEditMode) {
-          tooltip = isTransparentAndInvalid
-            ? __(
-                'Page background colors cannot have an opacity.',
-                'web-stories'
-              )
-            : __('Gradient not allowed for Text', 'web-stories');
-        }
+      <SwatchList role="listbox" aria-labelledby={ariaLabelledBy}>
```

Comment:
Nested `SwatchList` is a bit confusing 🤔

Reply:
Indeed, this also changes the style and adds extra space that's not needed. Looks like both the `role` and the `aria-labelledby` were assigned to the `SwatchList` already before. Curious which part of the tests required adding duplicate `SwatchList` here, perhaps the test can be adjusted instead.

Reply:
I guess you wrapped the options in additional `SwatchList` because `listbox` can't include a `button`?
Maybe create a new wrapper component instead of the original `SwatchList` instead? The original `SwatchList` should probably get the same props as before... So it would be something like
```
<Wrapper>
  <SwatchList ref={listRef} {...rest}>
  </SwatchList>
  <ColorAdd ... />
</Wrapper>
```
The `listRef` should still be on the `SwatchList`, otherwise navigation between colors with arrows will break.

Feel free to create a new issue for this particular accessibility issue and assign it to me -- doesn't look like that influences the e2e tests.

Reply:
>Curious which part of the tests required adding duplicate `SwatchList` here, perhaps the test can be adjusted instead.

The automated axe tests were failing once the test was fixed up because the `Add New Color` button was inside of the listbox. This is fine if there are other options in the listbox, but when there are no options (like in the e2e test for saved colors) then it fails the axe test because the listbox has no children with the role `group` or `option`.



> I guess you wrapped the options in additional `SwatchList` because `listbox` can't include a `button`? Maybe create a new wrapper component instead of the original `SwatchList` instead (or change the button to an `option`)? The original `SwatchList` should probably get the same props as before... So it would be something like
> 
> ```
> <Wrapper>
>   <SwatchList ref={listRef} {...rest}>
>   </SwatchList>
>   <ColorAdd ... />
> </Wrapper>
> ```
> 
> The `listRef` should still be on the `SwatchList`, otherwise navigation between colors with arrow keys will not work.
> 
> Feel free to create a new issue for this particular accessibility issue and assign it to me if it starts taking too much time -- doesn't look like that influences the e2e tests.

I think this is a great idea - this should work perfectly and then we wouldn't have to peel `aria-labelledby` off of the props too.

Reply:
Thanks for the suggestion! Code has been udpated.